dev/com.ibm.ws.repository.resolver/src/com/ibm/ws/repository/resolver/RepositoryResolver.java

```diff
@@ -82,13 +82,17 @@
 
     /**
      * Map from symbolic name to feature for all features returned as resolved by the kernel resolver
+     * <p>
+     * Keyset is mutually exclusive with {@link #featuresMissing} and {@link #requirementsFoundForOtherProducts}
      */
     Map<String, ProvisioningFeatureDefinition> resolvedFeatures;
 
     /**
      * List of requested features that were reported missing by the kernel resolver and weren't found in the repository applicable to another product.
      * <p>
      * May include sample names if they were missing
+     * <p>
+     * Mutually exclusive with {@link #resolvedFeatures} and {@link #requirementsFoundForOtherProducts}
      */
     List<String> featuresMissing;
```

Comment:
It's easy to validate that these sets (and `requirementsFoundForOtherProducts`) are mutually exclusive as they only have items added to them within the `resolveFeatures()` method.

The kernel resolver returns a list of resolved features and a list of missing features, and then the missing features are split between the `featuresMissing` and `requirementsFoundForOtherProducts` sets.